File: docs/System/Timer.js
export class Timer {
    constructor() {
        this.counter = 0;
    }

    wait(sec) {
        this.counter = sec * 60;
    }

    tick() {
        if(this.counter > 0){
            this.counter -= 1;
        }
    }

    is_waiting(){
        return this.counter != 0;
    }

    stop(){
        this.counter = 0;
    }
}

File: szakdoliV3/System/Timer.py
# -*- coding: utf-8 -*-

class Timer:

    def __init__(self):
        self.counter = 0

    def wait(self, sec):
        self.counter = int(sec * 60)

    def tick(self):
        if self.counter > 0:
            self.counter -= 1

    def is_waiting(self):
        return self.counter != 0

    def stop(self):
        self.counter = 0